### DISCUSSION TITLE
speaking in foreign language

### DISCUSSION THREAD

**Discussion by LucasZNK** [ggml-org/whisper.cpp#242] [Q&A] [answered]
Hi guys, im trying to transcribe audio from different language into text, only is working with english, what is the process to being able to do this? im receiving something like this

(speaking in foreign language)
(speaking in foreign language)

with spanish, french etc.. 

i installed the whisper.cpp , do make medium , ./main -f myfileinspanish.wav

**Comment by misutoneko** [comment]
You need to define the language, add "-l es" for Spanish.
Btw the original Python version does the language detection automatically, so this switch isn't needed there.

**Reply by Mickeyha** [reply]
@misutoneko I would like to inquire, can a model with a non-.en suffix automatically recognize languages within an Android application? Or do I need to pass parameters? Currently, I only know how to use it in Python.

**Reply by misutoneko** [reply]
Hi, I dont really know (I don't even own a smartphone of _any_ kind lol :)
But looking at the code, the language is set to "en" by default in app/src/main/jni/whisper/jni.c
(not sure if that's in effect).

**Comment by LucasZNK** [comment]
Yeah, this is why im asking i tried that, i need to download another model based on the language or something?

<img width="1194" alt="image" src="https://user-images.githubusercontent.com/57494138/206759515-65aa82d6-30db-45e8-948d-74e3d358409c.png">

**Comment by misutoneko** [comment, marked as answer]
I guess you get this message because the default model is english-only. So you need to define the model as well (the -m switch)

**Reply by LucasZNK** [reply]
You are right, thanks for the help. This works ./main -f audio-es-3.wav -m ./models/ggml-medium.bin -l es

**Comment by BrunoHenrique00** [comment]
@LucasZNK if you are looking for auto-detection of the language just use the flag  `-l 'auto'` when executing Whisper.
Hope this helps!
